### PR TITLE
Add CLI e2e test.

### PR DESCRIPTION
**Pull request by nazrhom**
This currently uses docker-compose to spin up the environment

### CODE REVIEW

File: test/EndToEnd/CLI.hs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    , configMinimumBidIncrement = fromJust $ intToNatural 8_000_000
+    }
+
+bidderBuysTest :: Assertion
+bidderBuysTest = runInDocker $ do
+  let seller = Alice
+      buyer1 = Bob
+      buyer2 = Carol
+
+  handleCliAction $ Prepare seller
```
Comment (uhbif19) [inline test/EndToEnd/CLI.hs:94]: Cool that it works, I thought there might be some problems in the way.

File: src/HydraAuction/Runner.hs [hunk 2]
```diff
@@ -172,6 +173,14 @@ executeTestRunner runner = do
           (MkExecutionContext {tracer = tracer, node = node, actor = Alice})
           runner
 
+executeDockerRunner :: Runner () -> IO ()
+executeDockerRunner runner = do
+  runProcess_ "make start-docker"
```
Comment (uhbif19) [inline src/HydraAuction/Runner.hs:225]: I think it better should stop `docker-compose` after. And also we could reuse same code for both docker cases.

**Review by uhbif19 — APPROVED** [review]

**Comment by uhbif19** [comment]
@MangoIV Please use that E2E stuff when doing Frontend CLI issue.

**Comment by uhbif19** [comment]
Inclusion test in test list is missing.

File: test/EndToEnd/HydraUtils.hs [hunk 3]
```diff
@@ -28,61 +29,74 @@ import CardanoNode (
...
+          let faucetTracer = contramap FromFaucet hydraTracer
+          seedFromFaucet_ node oscarCardanoVk 100_000_000 Fuel faucetTracer
+          seedFromFaucet_ node patriciaCardanoVk 100_000_000 Fuel faucetTracer
+          seedFromFaucet_ node rupertCardanoVk 100_000_000 Fuel faucetTracer
+          executeRunner ctx $ cont ((n1, Oscar), (n2, Patricia), (n3, Rupert))
 
 type ThreeClients =
   ((HydraClient, Actor), (HydraClient, Actor), (HydraClient, Actor))
 
-runningThreeNodesDockerComposeHydra ::
```
Comment (uhbif19) [inline test/EndToEnd/HydraUtils.hs:45]: Please do not remove this, I wanna to have ability to check tests on docker-compose locally.

Comment (nazrhom) [reply test/EndToEnd/HydraUtils.hs:45]: @uhbif19 I'm afraid this PR already breaks current docker setup, as it moves around some files in the `data` folder (this is needed, as the hydra nodes assumes a certain structure, you can only tell it the root folder in which to look for config files). We could definitely get the docker setup working again by updating the compose file and the scripts to respect the new structure, but it requires some extra effort.

Comment (uhbif19) [reply test/EndToEnd/HydraUtils.hs:45]: Okay.

Comment (nazrhom) [reply test/EndToEnd/HydraUtils.hs:45]: I created #273 to track this issue

File: test/EndToEnd/Hydra.hs [hunk 4]
```diff
@@ -113,9 +115,7 @@ testSuite =
     "L2"
     [ -- FIXME: Skiping.
       -- Fails because of hydra-node having different version than cabal lib.
```
Comment (uhbif19) [inline test/EndToEnd/Hydra.hs]:
```suggestion
    [
```

File: test/EndToEnd/HydraUtils.hs [hunk 5]
```diff
@@ -1,47 +1,113 @@
...
+  hydraKeysFor,
   hydraNodeActors,
+  keysFor,
  )
 
+spinUpHeads :: Int -> TxId -> (EmulatorDelegateClients -> Runner ()) -> Runner ()
+spinUpHeads clusterIx hydraScriptsTxId cont = do
+  liftIO $ do
+    hydraDir <- lookupProtocolParamPath
+    setEnv "HYDRA_CONFIG_DIR" hydraDir
```
Comment (uhbif19) [inline test/EndToEnd/HydraUtils.hs:66]: Please add comment on that.

Can it break whole terminal env?

File: src/HydraAuction/Runner.hs [hunk 6]
```diff
@@ -203,27 +207,29 @@ executeTestRunner runner = do
...
+dockerNode :: RunningNode
+dockerNode =
   RunningNode
     { networkId = Testnet $ NetworkMagic 42
     , nodeSocket = "./devnet/node.socket"
     }
 
-executeRunnerWithLocalNode :: forall x. Runner x -> IO x
-executeRunnerWithLocalNode runner = do
+executeDockerRunner :: Runner () -> IO ()
```
Comment (uhbif19) [inline src/HydraAuction/Runner.hs:223]: That it very strange function. `docker-compose` starts the whole cluster, not only L1 node.

File: src/HydraAuction/Runner.hs [hunk 7]
```diff
@@ -203,27 +207,29 @@ executeTestRunner runner = do
...
+executeDockerRunner runner = do
+  runProcess_ "make start-docker"
+  executeRunnerWithNodeAs dockerNode Alice runner
+
+executeRunnerWithNodeAs :: forall x. RunningNode -> Actor -> Runner x -> IO x
```
Comment (uhbif19) [inline src/HydraAuction/Runner.hs:228]: Actor would be removed for Runner in #248 anyway.

File: test/EndToEnd/HydraUtils.hs [hunk 8]
```diff
@@ -107,24 +173,28 @@ newtype DelegatesClusterEmulator a = DelegatesClusterEmulator
     )
 
 runEmulatorUsingDockerCompose :: DelegatesClusterEmulator a -> IO a
-runEmulatorUsingDockerCompose action =
-  runningThreeNodesDockerComposeHydra $ \clients -> do
-    intialStatesRef <-
+runEmulatorUsingDockerCompose action = runningThreeNodesDockerComposeHydra $ executeRunnerWithNodeAs dockerNode Alice . flip runEmulator action
+
+runEmulator :: EmulatorDelegateClients -> DelegatesClusterEmulator a -> Runner a
```
Comment (uhbif19) [inline test/EndToEnd/HydraUtils.hs:181]: :+1:

**Pull request merged**